r/tests/testthat/test-Array.R

```diff
@@ -260,18 +260,30 @@ test_that("array supports POSIXct (ARROW-3340)", {
   expect_array_roundtrip(times2, timestamp("us", "US/Eastern"))
 })
 
-test_that("array supports POSIXct without timezone", {
-  # Make sure timezone is not set
+test_that("array uses local timezone for POSIXct without timezone", {
   withr::with_envvar(c(TZ = ""), {
```

Comment:
Can we use `with_timezone` here too?

Reply:
It is erroring with `Error in assign(".sys.timezone", old, envir = base_env) : cannot change value of locked binding for '.sys.timezone'`. 
I think I am bumping into some cacheing issue.

> As from R 3.5.0, when a time zone location is first found in a session, its value is cached in object .sys.timezone in the base environment. 

AFAICT the caching works (as in it manages to unlock the `.sys.timezone` object, assign to it and then lock it again). The issue surfaces when we`return(tz)`. 

The ` withr::with_timezone(NA` chunk fails with the same error message when ran interactively.

Reply:
That is strange! The docs for withr don't _explicitly_ say this, but looking at the code it's "just" wrapping `with_envvar()` so `NAs` and `""` should both work there.

You should submit an issue to withr about this and see what they say. We can merge this without using it in these places, it's just a little less clean + might not be testing what we think depending on the sequence of the tests being run (e.g. if the cache is set before we get here, the `with_envvar()` isn't actually changing the timezone...